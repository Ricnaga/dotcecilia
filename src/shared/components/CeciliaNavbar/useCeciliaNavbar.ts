import { useNavigate } from 'react-router-dom';
import {
  AGREEMENT,
  CALCULATOR,
  LOGIN,
  PAYSTUB,
  ADVANCEPAYMENT,
} from '@application/routes/routes';
import { useLocalStorage } from '../../hooks/useLocalStorage';

export const useCeciliaNavbar = () => {
  const navigate = useNavigate();
  const { deleteUserData } = useLocalStorage();
  const links = [
    {
      href: CALCULATOR,
      children: 'Calculadora',
    },
    {
      href: PAYSTUB,
      children: 'Holerite',
    },
    {
      href: ADVANCEPAYMENT,
      children: 'Adiantamento',
    },
    {
      href: AGREEMENT,
      children: 'Acerto',
    },
  ];

  const logOut = () => {
    deleteUserData();
    navigate(LOGIN);
  };
  return {
    data: { links },
    functions: { logOut },
  };
};
