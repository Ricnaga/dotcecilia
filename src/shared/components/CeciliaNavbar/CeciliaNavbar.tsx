import { CeciliaButton } from '../CeciliaButton/CeciliaButton';
import { useCeciliaNavbar } from './hooks/useCeciliaNavbar';

export function CeciliaNavbar() {
  const {
    data: { links },
    functions: { logOut },
  } = useCeciliaNavbar();

  return (
    <div className="grid grid-cols-12 my-4">
      <nav className="col-span-11">
        {links.map(({ href, children }) => (
          <a key={children} href={href} className="mr-4">
            {children}
          </a>
        ))}
      </nav>
      <div className="col-span-1">
        <CeciliaButton title="Sair" className="py-1 text-lg" onClick={logOut} />
      </div>
    </div>
  );
}
