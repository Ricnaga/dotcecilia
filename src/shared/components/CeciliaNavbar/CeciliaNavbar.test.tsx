import { render } from '@testing-library/react';
import { CeciliaNavbar } from './CeciliaNavbar';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useNavigate: jest.fn(),
}));

describe('Component: CeciliaNavbar', () => {
  it('should render correctly', () => {
    const { container } = render(<CeciliaNavbar />);
    expect(container).toMatchSnapshot();
  });
});
