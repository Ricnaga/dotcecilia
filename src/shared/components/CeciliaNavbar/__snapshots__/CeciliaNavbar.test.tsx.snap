// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CeciliaNavbar should render correctly 1`] = `
<div>
  <div
    class="grid grid-cols-12 my-4"
  >
    <nav
      class="col-span-11"
    >
      <a
        class="mr-4"
        href="calculator"
      >
        Calculadora
      </a>
      <a
        class="mr-4"
        href="paystub"
      >
        Holerite
      </a>
      <a
        class="mr-4"
        href="advancepayment"
      >
        Adiantamento
      </a>
      <a
        class="mr-4"
        href="agreement"
      >
        Acerto
      </a>
    </nav>
    <div
      class="col-span-1"
    >
      <button
        class="btn btn-yellow py-1 text-lg"
        type="button"
      >
        Sair
      </button>
    </div>
  </div>
</div>
`;
